### ISSUE TITLE
Testing (development) - wrestler_and_opponent.txt - 1 

### ISSUE THREAD

**Issue by zhangyuanlin** [Topology/ALM-Compiler#15]
questions, discussion and code for entity-event-action module 
references: 
1) alm paper
2) Daniela's paper
3) Danlela's webpage for the library

**Comment by anuradha252** [comment]
Right, my bad. Agent is an attribute, and thus should be a function prefixed with event!
YL


Interesting. I actually don’t see it in the sort section. We have an event_agent
% Attribute Function [agent] for sort [event].
event_agent(#event, #entity).

So I wonder if this line

% Definition for function [defined_agent].
defined_agent(X) :- agent(X, Y).

should have been

% Definition for function [defined_agent].
defined_agent(X) :- event_agent (X, Y).
Anu


search the sort section to see if and how agent is defined there. 
YL

here is a small portion of the entity_event_and_action module

module entity_event_and_action
  sort declarations
    thing :: universe
    entity :: thing
    event :: thing
      attributes
        agent : entity -> booleans
        object : entity -> booleans   
    %actions :: event 
    activity :: event
  function declarations
    statics 
      defined
        defined_agent : event -> booleans
        defined_object : event -> booleans
    
  axioms
defined_agent(X) if agent(X, Y).
defined_object(X) if object(X, Y).

I don't understand why  agent(X, Y) in axioms when it id defined as  
agent : entity -> booleans


the exact error message
Exception in thread "main" java.lang.RuntimeException: SPARC  V2.54
: predicate agent of arity 2 at line 588, column 21 was not declared

line number 588 is
% Definition for function [defined_agent].
defined_agent(X) :- agent(X, Y).

**Comment by Topology** [comment]
fixed with #16.   If possible, can we keep 1 bug in 1 issue?

**Comment by zhangyuanlin** [comment]
Agree. This issue is NOT a bug report. It serves as a thread for discussion during test of this application.
